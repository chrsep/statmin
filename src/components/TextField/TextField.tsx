import styled from "../../styled-components"

export default styled.input`
  color: ${({ theme }) => theme.white};
  font-weight: bold;
  background: transparent;
  border: none;
  outline: 0;
  border-bottom: 2px solid ${({ theme }) => theme.accentColor};
  padding: 0.4rem 1px;
  transition: all 0.2s;
  opacity: 0.65;
  :focus {
    opacity: 1;
  }
`
