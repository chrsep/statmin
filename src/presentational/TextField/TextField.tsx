import * as React from "react"
import styled from "../../styled-components"

interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
  label?: string
  className?: string
  onChange?: (event: object) => any
}

export default ({ className, label, ...props }: Props) => (
  <TextFieldLabel className={className}>
    {label}
    <TextFieldInput {...props} />
  </TextFieldLabel>
)
const TextFieldLabel = styled.label`
  color: ${({ theme }) => theme.white};
  font-size: 1.4rem;
  display: flex;
  flex-flow: column;
  width: 100%;
`
const TextFieldInput = styled.input`
  color: ${({ theme }) => theme.white};
  font-weight: bold;
  background: transparent;
  border: none;
  outline: none;
  border-bottom: 2px solid ${({ theme }) => theme.white};
  padding: 0.4rem 1px;
  transition: all 0.2s;
  font-size: 1.8rem;
  opacity: 0.6;
  :focus {
    opacity: 1;
    border-bottom: 2px solid ${({ theme }) => theme.accentColor};
  }
`
