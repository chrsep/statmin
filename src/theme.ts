export default interface Theme {
  accentColor: string
  primaryColor: string
  white: string
  black: string
}

export const darkTheme: Theme = {
  accentColor: "#4fe6c4",
  black: "black",
  primaryColor: "black",
  white: "white"
}
