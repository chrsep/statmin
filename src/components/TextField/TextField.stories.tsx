import { action } from "@storybook/addon-actions"
import { storiesOf } from "@storybook/react"
import * as React from "react"
import { darkTheme } from "../../theme"
import TextField from "./TextField"

storiesOf("TextField", module).add("With label", () => (
  <TextField onClick={action("clicked")} theme={darkTheme} />
))
