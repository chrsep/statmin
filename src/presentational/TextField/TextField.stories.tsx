import * as React from "react"
import { storiesOf } from "storybook__react"
import { ThemeProvider } from "../../styled-components"
import { darkTheme } from "../../theme"
import TextField from "./TextField"

storiesOf("TextField", module).add("With label", () => (
  <ThemeProvider theme={darkTheme}>
    <TextField label="Personal access token" />
  </ThemeProvider>
))
