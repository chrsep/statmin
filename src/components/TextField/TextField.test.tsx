import { shallow } from "enzyme"
import "jest-styled-components"
import * as React from "react"
import { darkTheme } from "../../theme"
import TextField from "./TextField"

describe("Button Component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<TextField theme={darkTheme} />)
    expect(wrapper).toMatchSnapshot()
  })
})
