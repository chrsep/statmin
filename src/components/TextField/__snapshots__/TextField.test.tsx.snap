// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
.c0 {
  color: white;
  font-weight: bold;
  background: transparent;
  border: none;
  outline: 0;
  border-bottom: 2px solid #4fe6c4;
  padding: 0.4rem 1px;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  opacity: 0.65;
}

.c0:focus {
  opacity: 1;
}

<input
  className="c0"
/>
`;
